src/NBitcoin/Network.cs

```diff
@@ -10,1429 +12,1079 @@
+            set
+            {
+                EnsureNotFrozen();
+                this.litecoinWorkCalculation = value;
+            }
+        }
+
+        BigInteger proofOfStakeLimit;
+        public BigInteger ProofOfStakeLimit
+        {
+            get
+            {
+                return proofOfStakeLimit;
+            }
+            set
+            {
+                EnsureNotFrozen();
+                proofOfStakeLimit = value;
+            }
+        }
+
+        BigInteger proofOfStakeLimitV2;
+        public BigInteger ProofOfStakeLimitV2
+        {
+            get
+            {
+                return proofOfStakeLimitV2;
+            }
+            set
+            {
+                EnsureNotFrozen();
+                proofOfStakeLimitV2 = value;
+            }
+        }
+
+        int lastPOWBlock;
+        public int LastPOWBlock
+        {
+            get
+            {
+                return lastPOWBlock;
+            }
+            set
+            {
+                EnsureNotFrozen();
+                lastPOWBlock = value;
+            }
+        }
+
+        private bool frozen = false;
+        public void Freeze()
+        {
+            this.frozen = true;
+        }
+
+        private void EnsureNotFrozen()
+        {
+            if (this.frozen)
+            {
+                throw new InvalidOperationException("This instance can't be modified");
+            }
+        }
+
+        public virtual Consensus Clone()
+        {
+            var consensus = new Consensus();
+            this.Fill(consensus);
+            return consensus;
+        }
+
+        protected void Fill(Consensus consensus)
+        {
+            consensus.EnsureNotFrozen();
+            consensus.bIP34Hash = this.bIP34Hash;
+            consensus.hashGenesisBlock = this.hashGenesisBlock;
+            consensus.majorityEnforceBlockUpgrade = this.majorityEnforceBlockUpgrade;
+            consensus.majorityRejectBlockOutdated = this.majorityRejectBlockOutdated;
+            consensus.majorityWindow = this.majorityWindow;
+            consensus.minerConfirmationWindow = this.minerConfirmationWindow;
+            consensus.powAllowMinDifficultyBlocks = this.powAllowMinDifficultyBlocks;
+            consensus.powLimit = this.powLimit;
+            consensus.powNoRetargeting = this.powNoRetargeting;
+            consensus.powTargetSpacing = this.powTargetSpacing;
+            consensus.powTargetTimespan = this.powTargetTimespan;
+            consensus.ruleChangeActivationThreshold = this.ruleChangeActivationThreshold;
+            consensus.subsidyHalvingInterval = this.subsidyHalvingInterval;
+            consensus.minimumChainWork = this.minimumChainWork;
+            consensus.GetPoWHash = this.GetPoWHash;
+            consensus.coinType = this.CoinType;
+            consensus.litecoinWorkCalculation = this.litecoinWorkCalculation;
+            consensus.LastPOWBlock = this.LastPOWBlock;
+            consensus.ProofOfStakeLimit =
```

Comment:
that should probably be just initialized in a constructor and be { get; private set; } without having a property and field separated